src/Faker/Provider/ar_MA/Address.php

```diff
@@ -0,0 +1,358 @@
+        'سيدي بوعثمان',
+        'اليوسفية',
+        'الشماعية',
+        'العطاوية',
+        'آيت داوود',
+        'الحنشان',
+        'تالمست',
+        'تمنار',
+        'الرشيدية',
+        'ورززات',
+        'ميدلت',
+        'تنجداد',
+        'تنغير',
+        'أرفود',
+        'مولاي علي شريف',
+        'كلميمة',
+        'بوذنيب',
+        'أكدز',
+        'زاكورة',
+        'الريش',
+        'تازناخت',
+        'بومالن دادس',
+        'قلعة مغونة',
+        'أكادير',
+        'آيت ملول',
+        'الدشيرة الجهادية',
+        'إنزكان',
+        'القليعة',
+        'آيت ايعزة',
+        'الكردان',
+        'إيغرم',
+        'أولاد برحيل',
+        'أولاد تايمة',
+        'تاليوين',
+        'تارودانت',
+        'تافراوت',
+        'تزنيت',
+        'طاطا',
+        'آيت باها',
+        'بيوكرة',
+        'أقا',
+        'فم الحصن',
+        'فم زكيد',
+        'كلميم',
+        'طانطان',
+        'سيدي إفني',
+        'آسا',
+        'بويزكارن',
+        'الزاك',
+        'الوطية',
+        'الاخصاص',
+        'العيون',
+        'السمارة',
+        'بوجدور',
+        'طرفاية',
+        'المرسى',
+        'الداخلة',
+        'الكويرة',
+    );
+
+    protected static $buildingNumber = array('#####', '####', '##');
+
+    protected static $postcode = array('#####', '#####-####');
+
+    /**
+     * @link http://ar.wikipedia.org/wiki/%D9%85%D9%84%D8%AD%D9%82:%D9%82%D8%A7%D8%A6%D9%85%D8%A9_%D8%A7%D9%84%D9%88%D9%84%D8%A7%D9%8A%D8%A7%D8%AA_%D8%A7%D9%84%D8%A3%D9%85%D8%B1%D9%8A%D9%83%D9%8A%D8%A9_%D8%AD%D8%B3%D8%A8_%D8%A7%D9%84%D9%85%D8%B3%D8%A7%D8%AD%D8%A9
+     */
+    protected static $state = array(
+        'آيوا', 'أركنساس', 'أريزونا', 'ألاباما', 'ألاسكا', 'أوريغون', 'أوكلاهوما', 'أوهايو', 'أيداهو', 'إلينوي', 'إنديانا', 'الاباما', 'الجزر العذراء الأمريكية',
+        'بنس    يلفانيا', 'بورتو ريكو',
+        'تكساس', 'تينيسي',
+        'جزر ماريانا الشمالية', 'جورجيا',
+        'داكوتا الجنوبية', 'داكوتا الشمالية', 'ديلاوير', 'رود آيلاند',
+        'ساموا الأمريكية',
+        'غوام',
+        'فرجينيا الغربية', 'فلوريدا', 'فيرجينيا', 'فيرجينيا الغربية', 'فيرمونت',
+        'كارولاينا الجنوبية', 'كارولاينا الشمالية','كارولينا الشمالية', 'كاليفورنيا', 'كانساس', 'كنتاكي', 'كولورادو', 'كونيتيكت',
+        'لويزيانا',
+        'ماريلاند', 'ماساتشوستس', 'ماين', 'مسيسيبي', 'مونتانا', 'ميريلاند', 'ميزوري', 'ميشيغان', 'مين', 'مينيسوتا',
+        'نبراسكا', 'نيفادا', 'نيو جيرسي', 'نيو ميكسيكو', 'نيوهامشير', 'نيويورك',
+        'هاواي',
+        'واشنطن', 'وايومنغ', 'ويسكنسن', 'يوتا',
+    );
+
+    protected static $stateAbbr = array(
+       'AK', 'AL', 'AR', 'AZ', 'CA', 'CO', 'CT', 'DC', 'DE', 'FL', 'GA', 'HI', 'IA', 'ID', 'IL', 'IN', 'KS', 'KY', 'LA', 'MA', 'MD', 'ME', 'MI', 'MN', 'MO', 'MS', 'MT', 'NC', 'ND', 'NE', 'NH', 'NJ', 'NM', 'NV', 'NY', 'OH', 'OK', 'OR', 'PA', 'RI', 'SC', 'SD', 'TN', 'TX', 'UT', 'VA', 'VT', 'WA', 'WI', 'WV', 'WY'
+    );
+
+    /**
+     * @link http://www.nationsonline.org/oneworld/countrynames_arabic.htm
```

Comment:
the content of this site is copyrighted, we can't include it in Faker, which is MIT licensed. Please find another source.

Reply:
ok